pkg/streamingpromql/functions/rate_increase.go

```diff
@@ -114,7 +114,7 @@ func histogramRate(isRate bool, step types.RangeVectorStepData, hHead []promql.H
 				return histogram.ErrHistogramsIncompatibleSchema
 			}
 
-			if p.H.Schema < currentSchema {
+			if p.H.Schema < delta.Schema {
```

Comment:
I think we can avoid doing this so often. If we track the min schema, we only need to copy delta to the min schema once it has finished accumulating points (and /maybe/ on line 108 before the add (I'm not as sure about there, would need to check).

Reply:
Very possibly, but I'd like to keep the scope of this PR as small as possible, so I'll do that in a follow-up PR.

Reply:
https://github.com/grafana/mimir/pull/9531